src/js/core/directives/ui-grid-menu.js

```diff
@@ -46,6 +46,15 @@ function ($compile, $timeout, $window, $document, gridUtil, uiGridConstants, i18
 
       $scope.dynamicStyles = '';
 
+      if (uiGridCtrl && uiGridCtrl.grid.options.gridMenuTemplate) {
```

Comment:
I think your issue with the unit tests is that your if statement should be more detailed on the checks. Try changing it to the following:

```
if (uiGridCtrl && uiGridCtrl.grid && uiGridCtrl.grid.options && uiGridCtrl.grid.options.gridMenuTemplate) {
```

It may seem a bit overkill, but it will avoid any issues of the style, "Can't find blank of undefined"